File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int count = 0, element = nums[0];
        for (int i = 0; i < nums.length; i++) {
            if (count == 0) {
                element = nums[i];
            }

            if (nums[i] == element) {
                count += 1;
            } else {
                count -= 1;
            }
        }
        return element;
    }
}

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        max_count, element = 1, nums[0]
        numsSet = set(nums)
        for i in numsSet:
            count = nums.count(i)
            if max_count < count:
                max_count = count
                element = i
        else:
            return element
